packages/lib/WebDavApi.js

```diff
@@ -121,13 +121,13 @@ class WebDavApi {
 			attrValueProcessors: [attrValueProcessor],
 		};
 
-		return new Promise((resolve) => {
+		return new Promise((resolve, reject) => {
 			parseXmlString(xml, options, (error, result) => {
 				if (error) {
-					resolve(null); // Error handled by caller which will display the XML text (or plain text) if null is returned from this function
-					return;
+					reject(error);
```

Comment:
What would the error message be in this case? Because what sometimes happen is that the WebDAV service will return random data, like HTML or a plain text and in that case the parsing error is not useful. Like if the service returns a plain string "disk quota exceeded", it's not useful to display "parsing error at character 1", but sending back the full string, as we currently do, might help understand the problem.

The thing is that we can't expect getting anything sensible from the various WebDAV implementations. They are supposed to always return XML but in practice there are many poor implementations that return whatever data on failure.

Reply:
> sending back the full string, as we currently do

the string is actually truncated here
https://github.com/laurent22/joplin/blob/3c9b4b37bfe665cad0d66a7b9196774ffbf9a926/packages/lib/WebDavApi.js#L413
which sometimes makes is impossible to see the root cause as was the case in the linked issue.

But I see your point too, parsing error alone might not be enough. Also having full response text might be useful if for instance this is managed WebDAV server to show it to support. Maybe I should just remove that limit instead?

Reply:
We still need a limit because some WebDAV servers are badly written and send back just about anything. It could be a few bytes or it could a be 10 MB HTML blob, so if we don't restrict it it would make the error handling or logging fail and we want to avoid that.

Reply:
Agree. Not sure what's the best way. Whatever limit we put here, it may be not enough in some situations.